src/cml.js

```diff
@@ -159,9 +181,63 @@ class CML {
       ? ''
       : '![CML watermark](https://raw.githubusercontent.com/iterative/cml/master/assets/watermark.svg)';
 
-    const report = `${userReport}\n\n${watermark}`;
+    let userReport = testReport;
+    try {
+      if (!userReport)
+        userReport = await fs.promises.readFile(markdownFile, 'utf-8');
+    } catch (err) {
+      if (!watch) throw err;
+    }
+
+    let report = `${userReport}\n\n${watermark}`;
     const drv = getDriver(this);
 
+    const publishLocalFiles = () => async (tree) => {
+      const nodes = [];
+
+      visit(tree, ['definition', 'image', 'link'], (node) => nodes.push(node));
+
+      const visitor = async (node) => {
+        if (node.url && node.alt !== 'CML watermark') {
+          if (!triggerFile) watcher.add(node.url);
+          try {
+            const url = new URL(
+              await this.publish({ ...opts, path: node.url, session })
+            );
+            url.searchParams.set('cache-bypass', uuid.v4());
+            node.url = url.toString();
+          } catch (err) {
+            if (err.code !== 'ENOENT') throw err;
+          }
+        }
+      };
+
+      await Promise.all(nodes.map(visitor));
+    };
+
+    if (publish)
+      report = String(await remark().use(publishLocalFiles).process(report));
+
+    if (watch) {
+      let lock;
+      watcher.add(triggerFile || markdownFile);
+      watcher.on('all', async (event, path) => {
+        if (lock) return;
+        lock = true;
+        try {
+          winston.info(`watcher event: ${event} ${path}`);
+          await this.commentCreate({ ...opts, update: true, watch: false });
+          if (event !== 'unlink' && path === triggerFile)
+            await fs.promises.unlink(triggerFile);
+        } catch (err) {
+          winston.warn(err);
+        }
+        lock = false;
+      });
+      winston.info('watching for file changes...');
+      await new Promise((resolve) => resolve); // wait forever
```

Comment:
can not be done with a common `while`?

Also, is the watch mechanism UI blocker? Can I launch `send-comment --watch` and then continue?

Reply:
> _can not be done with a common `while`?_

Do you mean that some kind of infinite loop could be a better way of waiting forever? 🤔

Reply:
> _`while(true)`_

But that is [busy waiting](https://en.wikipedia.org/wiki/Busy_waiting) and would block the event loop! 🙀

Reply:
No. Thats the common loop in a game

Reply:
Events still occurs

Reply:
You can use in any case as a SetInterval which is nicer

Reply:
How can I achieve the same effect with `SetInterval` (?)

Reply:
In any case... the whole code flow is a bit weird

Reply:
That isn't functionally equivalent. The current code just pauses the execution; what you propose doesn't pause the execution and prints a text every second. 🙃

Reply:
Did you mean [`wait-foerever`](https://www.npmjs.com/package/wait-forever) by any chance? :sweat_smile:

Reply:
That's precisely what the `wait-forever` package does.

Reply:
Yes, one or the other, I do not care while we have a easy to understand name

Reply:
Resolved with https://github.com/iterative/cml/pull/1026/commits/a54bd354ecb09358d8e0e7e2ff97c3d29dfa9410

Reply:
Note that the previous code looked so dumb because `new Promise(_ => {})` wasn't allowed by the linter.